beater/config.go

```diff
@@ -42,26 +43,30 @@ type Config struct {
 	MaxConnections      int                    `config:"max_connections"`
 	MaxRequestQueueTime time.Duration          `config:"max_request_queue_time"`
 	Expvar              *ExpvarConfig          `config:"expvar"`
-	Frontend            *FrontendConfig        `config:"frontend"`
 	Metrics             *metricsConfig         `config:"metrics"`
 	AugmentEnabled      bool                   `config:"capture_personal_data"`
 	SelfInstrumentation *InstrumentationConfig `config:"instrumentation"`
+
+	RumConfig      *rumConfig `config:"rum"`
+	FrontendConfig *rumConfig `config:"frontend"`
+	rum            *rumConfig
```

Comment:
This should definitely work. It would however put logic that concerns the configuration file to all places using the configuration. 
I see handling the backwards compatible `FrontendConfig` as an implementation detail of the configuration, that is why I wanted to handle it directly at the config level. But I didn't find a way to do this without changing the signature for the frontendConfig, so I get your point.
I'm fine with changing when you think it makes it easier.

Reply:
I think the snippet above is simpler than https://github.com/elastic/apm-server/pull/1155/files#diff-5142cb122ddb0cf46ec3a72f64f52591R114, particularly knowing that eventually we'll remove the `frontend` and it would be a matter of removing such snippet and done.

We have some configuration logic in `beater.New` already (eg sourcemap read) so one `if` more should go unnoticed.

I was also wondering if I am missing something that would make it more complicated than what i think it is...

Reply:
I'm going to change it.